### DISCUSSION TITLE
How can I run a command with spaces constructed in an f-string?

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#4626] [General]
The following works:

```
vid='3.9'
$(f'python{vid}')
```

However

```
vid='3.9'
$(f'python{vid} -V')
```

does not work. The problem is the space. So I tried:

```
$([f'python{vid}', '-V'])
```

but that's not parsed properly for my purpose.

How can I run a command with spaces constructed in an f-string?

Many thanks for pointers!

**Comment by jnoortheen** [comment]
I think you should checkout the tutorial part again. Especially the subprocess mode and python mode parts. I know, it is confusing at the start. But the thing is, Xonsh is nothing but a Python REPL with some special handling for subprocess commands. 

The correct way to do the above is 

```py
# this captures the output and returns a string
$(@(f'python{vid}') @('-V'))

# or output to stdout
@(f'python{vid}') -V
```

**Reply by halloleo** [reply]
Thanks @jnoortheen ! 

Yes, of course, I can use `@` for commands as well! (I use `@` heavily for variables, but didn't realise that it works for commands too.)
